pyxrf/model/fileio.py

```diff
@@ -749,34 +809,90 @@ def render_data_to_gui(runid):
         id number for given run
     """
 
+    spectrum_cut = 3000  # Constant: the number of spectrum points to load 3000 ~ 3 kEv
+
     data_sets = OrderedDict()
     img_dict = OrderedDict()
-    fname = 'scan2D_{}'.format(runid)
 
-    data_out = fetch_data_from_db(runid)
-
-    # Transfer to standard format pyxrf GUI can take
-    fname_sum = fname+'_sum'
+    data_from_db = fetch_data_from_db(runid,
+                                      # Always create unique file name by adding
+                                      #   version number
+                                      fname_add_version=True,
+                                      # Always load data from all detectors
+                                      create_each_det=True,
+                                      # Always create data file (processing results
+                                      #   are going to be saved in the file)
+                                      output_to_file=True)
+
+    if not len(data_from_db):
+        logger.warning(f"No detector data was found in Scan #{runid}")
+        return
+    else:
+        logger.info(f"Data from {len(data_from_db)} detectors were found in Scan #{runid}.")
+        if len(data_from_db) > 1:
+            logger.warning(f"Selecting only the first dataset from Scan #{runid}.")
+
+    # If the experiment contains data from multiple detectors (for example two separate
+    #   Xpress3 detectors) that need to be treated separately, only the data from the
+    #   first detector is loaded. Data from the second detector is saved to file and
+    #   can be loaded from the file. Currently this is a very rare case (only one set
+    #   of such experiments from SRX beamline exists).
+    data_out = data_from_db[0]['dataset']
+    fname = data_from_db[0]['file_name']
+    detector_name = data_from_db[0]['detector_name']
```

Comment:
Not directly. The second dataset is saved as a separate file and this file can be opened from the disk. Typically the second dataset is processed separately anyway. The way the program is designed, there is no place to keep the second dataset and choose between two. Also there is no easy way to ask the user which dataset to load. I think that the future software should be written so that it supports simultaneous viewing and processing of multiple documents in different tabs. The user may be asked whether to open datasets as in separate tabs or just use them as channels. Also, multiple datasets from the same experiment may be kept in a single file (if working with files).

Reply:
Actually, there are only a few experiments currently exist that contain data from multiple detectors. You know one experiment at SRX (but they recorded only a test file with two detectors, no actual experimental). There are also a few old stepscans from SRX that have data from two detectors.

Reply:
OK, let's keep it "as is" for now, as it's going to be a major redesign of the program.